src/main/java/GameFactory.java

```diff
@@ -0,0 +1,50 @@
+public class GameFactory {
```

Comment:
We need to be careful with naming, a Java *Factory class I'd expect to instantiate different types instead of coordinating like this one does. I'd suggest naming this class differently.